File: iOS/LiveStreamCore/Sources/State/UserState.swift
//
//  UserState.swift
//  LiveStreamCore
//
//  Created by jeremiawang on 2024/10/22.
//

import RTCRoomEngine

public struct UserState: State {
    public var selfInfo: TUIUserInfo = TUIUserInfo()
    public var hasAudioStreamUserList: Set<String> = []
    public var hasVideoStreamUserList: Set<String> = []
    
    public init() {}
}

File: Android/tuilivekit/livestreamcore/src/main/java/com/trtc/uikit/livekit/livestreamcore/state/UserState.java
package com.trtc.uikit.livekit.livestreamcore.state;

import com.tencent.cloud.tuikit.engine.room.TUIRoomDefine.UserInfo;
import com.trtc.tuikit.common.livedata.LiveData;

import java.util.LinkedHashSet;

public class UserState {
    public UserInfo                        selfInfo               = new UserInfo();
    public LiveData<LinkedHashSet<String>> hasAudioStreamUserList = new LiveData<>(new LinkedHashSet<>());
    public LiveData<LinkedHashSet<String>> hasVideoStreamUserList = new LiveData<>(new LinkedHashSet<>());

    public void reset() {
        hasAudioStreamUserList.get().clear();
        hasVideoStreamUserList.get().clear();
    }
}
